### PR TITLE
[FIX](function) fix size function for array map

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes

Issue Number: close #xxx
now we use ```select size(map(1, 2));```  which will make be core 
and we can make size function handle array & map column both

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

File: regression-test/suites/query_p0/sql_functions/size_funciton/test_size_function.groovy [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    qt_sql """select size(array("2020-01-02", "2022-01-03", "2021-01-01", "1996-04-17")), array_shuffle(array("2020-01-02", "2022-01-03", "2021-01-01", "1996-04-17"), 0), shuffle(array("2020-01-02", "2022-01-03", "2021-01-01", "1996-04-17"), 0)"""
+    qt_sql "SELECT array_size(array_shuffle(['aaa', null, 'bbb', 'fff'])), array_shuffle(['aaa', null, 'bbb', 'fff'], 0), shuffle(['aaa', null, 'bbb', 'fff'], 0)"
+    qt_sql """select array_size(array("2020-01-02", "2022-01-03", "2021-01-01", "1996-04-17")), array_shuffle(array("2020-01-02", "2022-01-03", "2021-01-01", "1996-04-17"), 0), shuffle(array("2020-01-02", "2022-01-03", "2021-01-01", "1996-04-17"), 0)"""
+    qt_sql "SELECT (cardinality(['aaa', null, 'bbb', 'fff'])), array_shuffle(['aaa', null, 'bbb', 'fff'], 0), shuffle(['aaa', null, 'bbb', 'fff'], 0)"
+    qt_sql """select cardinality(array("2020-01-02", "2022-01-03", "2021-01-01", "1996-04-17")), array_shuffle(array("2020-01-02", "2022-01-03", "2021-01-01", "1996-04-17"), 0), shuffle(array("2020-01-02", "2022-01-03", "2021-01-01", "1996-04-17"), 0)"""
+
+    qt_sql "SELECT size(map(1,2,2,null));"
```
Comment (morrySnow) [inline regression-test/suites/query_p0/sql_functions/size_funciton/test_size_function.groovy:29]: please add signature `size(map)` for Nereids

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**